seatunnel-formats/seatunnel-format-json/src/main/java/org/apache/seatunnel/format/json/canal/CanalJsonDeserializationSchema.java

```diff
@@ -113,92 +113,97 @@ public SeaTunnelDataType<SeaTunnelRow> getProducedType() {
         return this.physicalRowType;
     }
 
-    @Override
-    public void deserialize(byte[] message, Collector<SeaTunnelRow> out) {
-        if (message == null) {
-            return;
-        }
-        ObjectNode jsonNode = (ObjectNode) convertBytes(message);
-        assert jsonNode != null;
-        deserialize(jsonNode, out);
-    }
-
-    public void deserialize(ObjectNode jsonNode, Collector<SeaTunnelRow> out) {
-        if (database != null
-                && !databasePattern.matcher(jsonNode.get(FIELD_DATABASE).asText()).matches()) {
-            return;
-        }
-        if (table != null && !tablePattern.matcher(jsonNode.get(FIELD_TABLE).asText()).matches()) {
-            return;
-        }
-        JsonNode dataNode = jsonNode.get(FIELD_DATA);
-        String type = jsonNode.get(FIELD_TYPE).asText();
-        // When a null value is encountered, an exception needs to be thrown for easy sensing
-        if (dataNode == null || dataNode.isNull()) {
-            // We'll skip the query or create or alter event data
-            if (OP_QUERY.equals(type) || OP_CREATE.equals(type) || OP_ALTER.equals(type)) {
+    public void deserialize(ObjectNode jsonNode, Collector<SeaTunnelRow> out) throws IOException {
+        try {
+            if (database != null
+                    && !databasePattern.matcher(jsonNode.get(FIELD_DATABASE).asText()).matches()) {
                 return;
             }
-            throw new SeaTunnelJsonFormatException(
-                    CommonErrorCodeDeprecated.JSON_OPERATION_FAILED,
```

Comment:
@Hisoka-X Which new one can replace `CommonErrorCodeDeprecated.JSON_OPERATION_FAILED`?

Reply:
I think you should created new one in `CommonErrorCode` to instead of this one. You can follow exist `CommonErrorCode`.

Reply:
Thanks for your reminder.